### ISSUE TITLE
Deferred (execution) rendering

### ISSUE THREAD

**Issue by eliias** [eliias/isorun#47]
# Problem

When multiple applications are rendered on the same view, the rendering is not going to happen in parallel (and also not concurrently), but are a sequence of function calls. This results in an undesired waterfall like execution. We release the GVL for almost the whole render time (except calls to Ruby), and we could easily multi-thread rendering as well.

## How

A prerequisite for deferred rendering is the possibility to make non-blocking calls to the render functions and wait for the results at the end. Rails views do not offer any concurrency patterns or deferred rendering of helpers. We can still achieve this in the following ways:
- Create the container node immediately (i.e. `<div id="my_app"/>`)
  - Wait for all _async_ parts to eventually resolve before returning the view
  - Push the results of the call (like _turbo stream_ does) as soon as they are ready
